### ISSUE TITLE
Let users configure the "Z" flag when mouting a volume

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#2874]
This is a feature that has been discussed with @eivantsov last week and should help solving some problems that customers may have with volumes mounted with the Z flag.

Currently wsmaster add the Z flag when mounting volumes. For example this command is used to start a java workspace:

`docker create -v /home/user/che/workspaces/wksp-sr4g:/projects:Z ... docker.io/codenvy/ubuntu_jdk8`

The problem with the Z flag is that there are situations where the container creation [will fail if it's set](https://github.com/docker/docker/issues/17262). This is currently the case of CentOS 7 with SELinux activated for example.

The proposed solution is to introduce a new property (e.g. `machine.docker.use_zflag`) to allow users to disable the Z flag. The default behaviour will be to set the Z flag.

**Comment by TylerJewell** [comment]
Since there can be many flags applied to a mount how about just a string that users can give that would override the current flag with a new one?

**Comment by l0rd** [comment]
@snjeza is working on that

**Comment by l0rd** [comment]
@TylerJewell that's a good point. Instead of true/false the value could be "Z" or "z" or whatever flag we want to apply to all volumes.

**Comment by TylerJewell** [comment]
Or :ro for example.  We may want to also let users ask which flags are applied by default too?  Output that in the info command.

**Comment by l0rd** [comment]
@TylerJewell currently when starting a new workspace the following volumes are mounted:

```
-v /home/user/che/lib/linux_amd64/terminal:/mnt/che/terminal:ro,Z
-v /home/user/che/lib/ws-agent.tar.gz:/mnt/che/ws-agent.tar.gz:ro,Z
-v /home/user/che/workspaces/wksp-92fb:/projects:Z
```

If the new property we want to introduce has `:ro` as value, would it make sense to set `ro` to all volumes?

**Comment by TylerJewell** [comment]
I wouldn't think so. For example, the product wouldn't work at all if you applied that property to /projects.  So given that :Z would be applied to all volumes and :ro would not be, then maybe we do have to go back to the idea that each property type has a seprate variable for it.  And we have to be more selective so that users can set Z value or RO separately.

**Comment by JamesDrummond** [comment]
@l0rd Can you label this kind enhancement for tracking please.

**Comment by l0rd** [comment]
@JamesDrummond done :-)

**Comment by l0rd** [comment]
I'm closing this issue since it has been fixed by #2921
